### ISSUE TITLE
Зачем полностью генерировать столько кода

### ISSUE THREAD

**Issue by dimzon** [prickly-u/CompactSerializer#1]
Все проще - прочитали значение свойства на вершину стека, позвали соответствующий метод System.IO.BinaryWriter
Для тех типов которые не поддерживаются BinaryWriter напрямую делаем хелпер:

```
public static class BinaryWriterHelper{
   public static void WriteDateTime(BinaryWriter writer,  DateTime value){
     // Нафига эмитить для каждого свойства код если его можно в статический метод написать руками а потом просто эмитить вызов статического метода? 
   }
}

```

**Comment by foghegehog** [comment]
Да, для простых типов он действительно делает по факту тоже самое, что эмиченный код. Вызов статического метода да и вообще самих методов BinaryWriter из системной сборки может быть немного дольше, чем inplace запись, но может они будут инлайниться, а может это вовсе экономия на спичках и способ не настолько extra fast. Надо мерять, если будет время/возможность - попробую.

**Comment by dimzon** [comment]
И в догонку - перед тем как велосипедить:

https://github.com/aumcode/serbench
https://github.com/stebet/SerializationBenchmarks
https://github.com/cdorst/Benchmarks.EntitySerialization
https://github.com/fed135/buffer-zip-benchmarks
https://github.com/rpgmaker/NetJSON
https://github.com/Alois-xx/SerializerTests
https://aloiskraus.wordpress.com/2017/04/23/the-definitive-serialization-performance-guide/
http://aumcode.github.io/serbench/
